utils/output.py

```diff
@@ -0,0 +1,147 @@
+    got_output_normalized_path = got_output_path + ".normalized.yaml"
+    # create a preprocessed/normalized outfile file to allow its comparision across platforms and setups
+    with open(got_output_normalized_path, 'w') as f:
+            yaml.dump(normalize_output(got_output, input_root_path), f)
+    with open(got_output_normalized_path, encoding='utf-8') as file:
+        got_output = yaml.safe_load(file)
+
+    if not os.path.exists(expected_output_dir):
+        os.mkdir(expected_output_dir)
+
+    if not os.path.exists(expected_output_path):
+        with open(expected_output_path, 'w') as f:
+            yaml.dump(got_output, f)
+
+        assert False, "Expected output file '%s' did not exist, initializing it with the current test output" % got_output_normalized_path
+
+    else:
+        with open(expected_output_path) as f:
+            expected_output = yaml.safe_load(f)
+
+    assert got_output == expected_output, "Got different analysis output: \n%s" % get_files_diff(expected_output_path, got_output_normalized_path)
+
+
+def assert_analysis_output_dependencies(expected_output_dir, output_dir):
+    """
+    Asserts that the Dependencies from analysis output
+
+    Raises:
+        AssertionError: If dependencies were different.
+
+    Returns:
+        None.
+
+    """
+
+    expected_output, expected_output_path = dict(), os.path.join(expected_output_dir, "dependencies.yaml")
+    got_output, got_output_path = get_dict_from_output_file("dependencies.yaml", dir=output_dir)
+
+    if not os.path.exists(expected_output_path):
+        with open(expected_output_path, 'w') as f:
+            yaml.dump(got_output, f)
+
+        assert False, "Expected dependencies file '%s' did not exist, initializing it with the current test output" % expected_output_path
+
+    else:
+        with open(expected_output_path) as f:
+            expected_output = yaml.safe_load(f)
+
+    assert got_output == expected_output, "Got different dependencies output: \n%s" % get_files_diff(expected_output_path, got_output_path)
+
+
+def get_dict_from_output_file(filename, dir=None, **kwargs):
+    """
+        Loads and returns a YAML from analysis output Violation file output.yaml, TODO dir cleaner
+
+        Returns:
+            dict (from YAML file data)
+
+        """
+    report_path = os.getenv(constants.REPORT_OUTPUT_PATH)
+    report_path = kwargs.get('report_path', report_path)
+    if dir:
+        report_path = dir
+    output_path = os.path.join(report_path, filename)
+
+    with open(output_path, encoding='utf-8') as file:
+        return yaml.safe_load(file), output_path
+
+
+def normalize_output(rulesets: dict, input_root_path):
+    """
+        Does a pruning on output file to delete not used fields (skipped and unmatched rules),
+        makes incident paths generic to allow compare container and container-less results.
+    """
+    for ruleset in rulesets:
```

Comment:
This and following method tries to process analysis output to unify its results acoss platforms and container/less types of analysis, likely going to be updated with next added applications.